### PR TITLE
[WIP]Fixing state transition during processing

### LINKED ISSUE #204: SUSI stops listening while processing the queries
**Actual Behaviour**

SUSI stops listening while processing the queries,eg. while playing a song , when you say 'google stop' in google home , it stops the music and 'processes another query(stop) while still processing the first one(playing music)' ,
so actions like stop action do not work in SUSI right now. Since , it only processes one query

**Expected Behaviour**

To fix this bug

**Steps to reproduce it**

Try giving some command while SUSI is processing a previous query

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by stealthanthrax**
Fixes #204 
In this PR I am trying to shift to idle state from busy state , if the user uses the Hotword Again.
My current approach would be to use multi processes to start hotword detection again.

Suggestions for alternate approaches are welcome

### CODE REVIEW

**Comment by prateekiiest** [comment]
Any updates on this @stealthanthrax.

I see there has been some conflicts recently.

**Comment by stealthanthrax** [comment]
@prateekiiest , please can you give any suggestions on the approach to follow for this issue

File: main/states/busy_state.py [hunk 1]
```diff
@@ -4,13 +4,18 @@
...
-
+       
 class BusyState(State):
     """Busy state inherits from base class State. In this state, SUSI API is called to perform query and the response
     is then spoken with the selected Text to Speech Service.
     """
-
+    def tmkc(self):
+        if self.components.hotword_detector.on_detected():
+            os.system('play ' + '/home/pi/SUSI.AI/susi_linux/wav/infobleep.wav')
```
Comment (prateekiiest) [inline main/states/busy_state.py]: include `subprocess` here

File: main/states/busy_state.py [hunk 2]
```diff
@@ -23,12 +28,16 @@ def on_enter(self, payload=None):
...
             GPIO.output(27, True)
+            p1=Process(target = self.tmkc())
             if self.components.renderer is not None:
                 self.notify_renderer('speaking', payload={'susi_reply': reply})
-
             if 'answer' in reply.keys():
                 print('Susi:' + reply['answer'])
-                self.__speak(reply['answer'])
+                p2=Process(target=self.__speak,args=((reply['answer']),))
+                p2.start()
```
Comment (prateekiiest) [inline main/states/busy_state.py]: Why are you creating two threads here?

**Comment by stealthanthrax** [comment]
Not relevant anymore

**Pull request closed without merging**